tests/integration/anoncreds/derived-credentials.test.js

```diff
@@ -305,6 +306,116 @@ for (const {
       await createAndVerifyPresentation(credentials);
     }, 30000);
 
+    test('Holder creates a derived verifiable credential from a credential with range proofs', async () => {
+      const provingKeyId = 'provingKeyId';
+      const pk = BoundCheckSnarkSetup();
+      const provingKey = pk.decompress();
+
+      const issuerKey = getKeyDoc(did1, keypair, keypair.type, keypair.id);
+      const unsignedCred = {
+        ...credentialJSON,
+        issuer: did1,
+      };
+
+      const presentationOptions = {
+        nonce: stringToU8a('noncetest'),
+        context: 'my context',
+      };
+
+      // Create W3C credential
+      const credential = await issueCredential(issuerKey, unsignedCred);
+
+      // Begin to derive a credential from the above issued one
+      const presentationInstance = new Presentation();
+      const idx = await presentationInstance.addCredentialToPresent(
+        credential,
+        { resolver },
+      );
+
+      // NOTE: revealing subject type because of JSON-LD processing for this certain credential
+      // you may not always need to do this depending on your JSON-LD contexts
+      await presentationInstance.addAttributeToReveal(idx, [
+        'credentialSubject.type.0',
+      ]);
+      await presentationInstance.addAttributeToReveal(idx, [
+        'credentialSubject.type.1',
+      ]);
+
+      // Enforce LPR number to be between values aswell as revealed
+      // NOTE: unlike other tests, we cannot "reveal" this value and enforce bounds at the same time!
+      presentationInstance.presBuilder.enforceBounds(
+        idx,
+        'credentialSubject.lprNumber',
+        1233,
+        1235,
+        provingKeyId,
+        provingKey,
+      );
+
+      // Enforce issuance date to be between values
+      // NOTE: we dont need to set proving key value here (must still set ID though!) as its done above, should pass undefined
+      presentationInstance.presBuilder.enforceBounds(
+        idx,
+        'issuanceDate',
+        new Date('2019-10-01'),
+        new Date('2020-01-01'),
+        provingKeyId,
+        undefined,
```

Comment:
is there any drawback if we send the `provingKey` in all calls?

Reply:
it will throw an error

Reply:
I will need to fix that in the wallet then, thanks